### DISCUSSION TITLE
how to hide  QtWebEngineProcess.exe shell when use pyinstaller ？

### DISCUSSION THREAD

**Discussion by CatNofishing** [pyinstaller/pyinstaller#6191] [PyInstaller] [answered]
I use pyinstaller to package my program, but there is a QtWebEngineProcess.exe black shell window, how to hide it
![image](https://user-images.githubusercontent.com/33288114/132131188-1ae61365-c8f6-4462-bb6c-fc4e51e3cb5f.png)

**Comment by bwoodsend** [comment]
Sounds like you're just needing the [--windowed](https://pyinstaller.readthedocs.io/en/latest/usage.html#cmdoption-c) option.

**Comment by rokm** [comment, marked as answer]
Can you provide a minimum code example that causes the shell being opened like that when packaged with PyInstaller? What version of PyQt5 / PyQtWebEngine are you using?

I have not seen the QtWebEngineProcess.exe helper open shell window on its own before; it is usually built with GUI subsystem, so it has no console.

I suppose in a windowed/noconsole PyInstaller-frozen application, the console would be opened if a subprocess was launched with shell enabled, but I doubt QtWebEngineWidgets do that. And basic QtWebEngineView example does indeed not seem to open console when built and ran in noconsole mode...

**Reply by CatNofishing** [reply]
@rokm  @bwhite   I have used the -w option to cancel the console window, but the PyQtWebEngine process window still exists. My pyinstaller version is 4.5.1, and my module is 
pandasgui==0.2.10.1
numpy==1.21.2
QDarkStyle==3.0.0
matplotlib==3.4.3
mplcursors==0.4
scipy==1.7.1
pandas==1.3.2
PyQt5==5.15.4
.
This window comes from Pandasgui, I click a button, and then a child process will be started to execute pandasgui.show(df), and this window will appear after packaging. It should be noted that if I do not use the -w option when packaging, this window will not appear, once I use -w, this window will appear
![image](https://user-images.githubusercontent.com/33288114/132185065-6cb9da5a-ca1c-4925-871e-267bb0c0f6c0.png)


 This is the video of my operation before packaging， it work fine no any problem.
![GIF 2021-09-06 16-35-56](https://user-images.githubusercontent.com/33288114/132186851-aa097916-d97c-4804-b885-a3f3a996fb61.gif)

This is the running video after I packaged it with the pyinstaller -w -D command. You will find that when I click the menu bar button to start pandasgui, the QtWebEngineProcess.exe shell will appear。 Of course, there is another small problem, that is, my mouse will become busy, when I click the "能耗优化" button
![GIF 2021-09-06 16-40-56](https://user-images.githubusercontent.com/33288114/132187693-d20db43d-cb8c-4258-b18a-eef6d995a187.gif)

**Reply by rokm** [reply]
> It should be noted that if I do not use the -w option when packaging, this window will not appear, once I use -w, this window will appear.

That's because if you don't use the -w option, the application already has a console, so the launched sub-process (that requires console for whatever reason) does not open an additional console. If there is no console available in the application, the sub-process that requires a console will open one, which is what you are seeing.

---

Anyway, I don't think that is PyInstaller bug/problem, as it can be easily reproduced by running some `pandasgui` code in windowed python interpreter. For example, if you run

```
python.exe -m pandasgui.widgets.figure_viewer
```

from console, you get a window and no additional console (since application/python interpreter inherited the one from which it was launched).

But if you run in windowed interpreter (ending with 'w'):

```
pythonw.exe -m pandasgui.widgets.figure_viewer
```

you get the window and the QtWebEngineProcess.exe console window.

This behavior seems to be caused by [setting chromium debug flags here](https://github.com/adamerose/PandasGUI/blob/5b819d3333364c57be04fe5c8ad80ba2a340e22d/pandasgui/widgets/figure_viewer.py#L16) and can be suppressed by commenting out that line. 

Can you:
a) check that you get the same behavior (console window) when you run your application source with windowed python interpreter (`pythonw.exe`)?
b) check if it goes away if you comment out the above-referenced line in your local `pandasgui` installation?

**Reply by CatNofishing** [reply]
@rokm， amazing，it works now. Thanks your reply.  😁But ，there is another problem, it is that  my mouse will become busy, when I click the "能耗优化" button. This button will start a python thread thread, the thread through the process pool process complex time-consuming calculations.Before I package, everything is normal, once I packaged, click the button to start running, the mouse will turn into a busy state, but the interface is still Smooth, what is going on. I don’t understand why the state of the mouse is different in the same operation before and after packaging
![image](https://user-images.githubusercontent.com/33288114/132272771-16a8ab00-f3bd-4abe-b57e-5eca7ee116b6.png)

![GIF 2021-09-07 9-56-39](https://user-images.githubusercontent.com/33288114/132272596-85499c19-ea6d-44b6-947b-cea60f872b43.gif)

**Reply by rokm** [reply]
No idea about the busy indicator, sorry. Can you reproduce the problem with a simple stand-alone program (e.g., a window and a button that launches a busy thread when pressed)? Debugging complex applications without source is kind of beyond the scope of what we can do here...

**Reply by CatNofishing** [reply]
thank you for your help, 😁That’s a bit difficult. It’s difficult for me to present this problem with a standalone program. I will upload my code when I graduate.
